File: Java/Demo/Codility Demo Test.java
// https://app.codility.com/demo/take-sample-test/
import java.util.Arrays;

class Solution {
    public int solution(int[] A) {
        Arrays.sort(A);
        if (A[A.length - 1] < 1) {
            return 1;
        }

        int number = 0;
        int i = 0;
        while (i < A.length) {
            if (A[i] <= number) {
                i++;
            } else if (A[i] == number + 1) {
                number++;
                i++;
            } else {
                break;
            }
        }
        return number + 1;
    }
}

File: JavaScript/Demo/Codility Demo Test.js
// https://app.codility.com/demo/take-sample-test/
function solution (A) {
  A.sort((a, b) => a - b)

  let expectedInt = 1

  for (let i = 0; i < A.length; i++) {
    if (A[i] > expectedInt) {
      break
    } else if (A[i] === expectedInt) {
      expectedInt++
    }
  }

  return expectedInt
}